python/src/video_library.py

```diff
@@ -25,7 +32,7 @@ def __init__(self):
                 self._videos[url] = Video(
                     title,
                     url,
-                    tags.split(","),
+                    tuple(_split_tags_ignoring_whitespace(tags)),
```

Comment:
in _theory_ the tuple() is not needed here because we copy it into a tuple anyway in the Video Constructor.

Reply:
Or we could even change this to just be like [tag.strip() for tag in tags.split(",") if tag], WDYT?

Reply:
I chose `tuple` to change the type from `Iterator` to `Sequence` to conform to the type annotation in `Video`'s constructor. Instead of collecting the yielded entries in a tuple, we could also change the type annotation in Video to `Iterable`.

I chose the current solution to handle an empty tag, e.g. `#first,  , #third`, but now realize that your proposed code and the current Java implementation would insert an empty tag in the case. Will update in order to match Java's behavior and simplify the code.

Reply:
hmmm both solutions should just be an empty list if there are no tags. like `[]` and not `['']`. The latter would be wrong and we'd need to fix it in both cases :)